kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -486,6 +494,87 @@ public ConsumerCommittedResponse committed(
     return response;
   }
 
+  /**
+   * Returns the beginning offset of the {@code topic} {@code partition}.
+   */
+  public long getBeginningOffset(String topic, int partition) {
+    log.debug("Beginning offset for topic {} and partition {}.", topic, partition);
+    KafkaConsumerState<?, ?, ?, ?> consumer;
+    try {
+      consumer =
+          retrying(
+              () -> getConsumerInstance(getAdminConsumerInstance()),
+              ConsumerInstanceNotFoundException.class,
+              CONSUMER_REQUEST_MAX_ATTEMPTS);
+    } catch (ConsumerInstanceNotFoundException e) {
+      throw new IllegalStateException("Unable to get hold of admin consumer instance.", e);
+    }
+    return consumer.getBeginningOffset(topic, partition);
+  }
+
+  /**
+   * Returns the end offset of the {@code topic} {@code partition}.
+   */
+  public long getEndOffset(String topic, int partition) {
+    log.debug("End offset for topic {} and partition {}.", topic, partition);
+    KafkaConsumerState<?, ?, ?, ?> consumer;
+    try {
+      consumer =
+          retrying(
+              () -> getConsumerInstance(getAdminConsumerInstance()),
+              ConsumerInstanceNotFoundException.class,
+              CONSUMER_REQUEST_MAX_ATTEMPTS);
+    } catch (ConsumerInstanceNotFoundException e) {
+      throw new IllegalStateException("Unable to get hold of admin consumer instance.", e);
+    }
+    return consumer.getEndOffset(topic, partition);
+  }
+
+  /**
+   * Returns the earliest offset whose timestamp is greater than or equal to the given {@code
+   * timestamp} in the {@code topic} {@code partition}, or empty if such offset does not exist.
+   */
+  public Optional<OffsetAndTimestamp> getOffsetForTime(
+      String topic, int partition, Instant timestamp) {
+    log.debug("Offset for topic {} and partition {} at timestamp {}.", topic, partition, timestamp);
+    KafkaConsumerState<?, ?, ?, ?> consumer;
+    try {
+      consumer =
+          retrying(
+              () -> getConsumerInstance(getAdminConsumerInstance()),
```

Comment:
I think `getConsumerState` would be a better name for this and we could consider changing it in this PR ...